wordpress-mobile/WordPressAuthenticator-iOS
# Update SIWA error message

Improvement for SIWA support in WCiOS https://github.com/woocommerce/woocommerce-ios/issues/2643

## Changes

Before this PR, we show a pretty generic error "Apple authentication failed." to the user. In case the user didn't realize that they need to sign in to iCloud with an Apple ID with 2 factor enabled, it'd be nice to remind them about these requirements. It doesn't look the prettiest since it's just a bunch of text, feel free to suggest other copy!

- Updated the user-facing error message in `AppleAuthenticator`'s error scenario
- I noticed that my changes to provide default strings to `WordPressAuthenticatorDisplayStrings`'s initializer in [an earlier PR](https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/pull/373/files) were reverted probably in a commit that fixes merge conflicts in [this PR](https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/pull/374/files). Since WCiOS does pass a small subset of custom strings to `WordPressAuthenticatorDisplayStrings`'s init, it'd be great if we keep these default strings to avoid a very long initializer in WCiOS. @mindgraffiti please lemme know if there are concerns with these default strings in the initializer!

## Testing

- In `Podfile`, update `WordPressAuthenticator` pod to
```
pod 'WordPressAuthenticator', :git => 'https://github.com/wordpress-mobile/WordPressAuthenticator-iOS.git', :branch => 'wcios-2643/update-apple-auth-error-message'
```
- Launch the app in a simulator/device that doesn't have an iCloud account
- Log out from the app
- Tap on "Log In" CTA
- Tap on "Continue with Apple" --> should see an alert modal as in the screenshots below

## Example screenshots

WPiOS:

dark | light
-- | --
![Simulator Screen Shot - iPhone 11 Pro - 2020-08-21 at 17 01 32](https://user-images.githubusercontent.com/1945542/90875405-b2777800-e3d3-11ea-935e-ef8c39bddc20.png) | ![Simulator Screen Shot - iPhone 11 Pro - 2020-08-21 at 17 02 10](https://user-images.githubusercontent.com/1945542/90875412-b60aff00-e3d3-11ea-933d-8df514f46d57.png)

WCiOS:

dark | light
-- | --
![Simulator Screen Shot - iPhone 11 Pro - 2020-08-21 at 17 19 30](https://user-images.githubusercontent.com/1945542/90875439-c15e2a80-e3d3-11ea-87ff-524c582c85f9.png) | ![Simulator Screen Shot - iPhone 11 Pro - 2020-08-21 at 17 19 40](https://user-images.githubusercontent.com/1945542/90875452-c4f1b180-e3d3-11ea-99e1-d9d9fd3ecccd.png)
